tests/meson.build

```diff
@@ -0,0 +1,18 @@
+gtest_main_dep = dependency('gtest', main: true, required: false)
+if not gtest_main_dep.found()
+  gtest_proj = subproject('gtest')
+  gtest_main_dep = gtest_proj.get_variable('gtest_main_dep')
+endif
```

Comment:
This seems complicated. A subproject fallback will occur automatically if the dependency is required, so you can do:

```meson
gtest_main_dep = dependency('gtest_main', 'gtest', main: true)
```

You need to specify both dependency names with the preferred one (gtest_main) first, so that you get a fallback to it via the wrap file. The `main: true` is ignored when trying `gtest_main`, but is used for `gtest` (so that meson will use its internal lookup method when pkg-config is not available). Long term, meson needs to deprecate that kwarg and consolidate on "gtest_main" because the current state of affairs is confusing...